mllib/src/main/scala/org/apache/spark/ml/classification/OneVsRest.scala

```diff
@@ -325,8 +350,13 @@ final class OneVsRest @Since("1.4.0") (
       multiclassLabeled.persist(StorageLevel.MEMORY_AND_DISK)
     }
 
+    val iters = Range(0, numClasses).par
+    iters.tasksupport = new ForkJoinTaskSupport(
+      new ForkJoinPool(Math.min(getParallelism, numClasses))
+    )
+
     // create k columns, one for each binary classifier.
-    val models = Range(0, numClasses).par.map { index =>
+    val models = iters.map { index =>
```

Comment:
They don't necessarily add anything to here, but they are a more standard way of parallelism in Spark over using `TaskSupport` and it's more flexible for setting an ExecutorService.  I'm not sure if you can set `TaskSupport` to `sameThreadExecutor` or what really happens behind the scenes if you make a `ThreadPoolTaskSupport` with 1 thread.